### PR TITLE
Fix hierarchy node styling

### PR DESCRIPTION
**Pull request by SYBIOTE**
This pull request fixes the styling of the HierarchyTreeNode component. It updates the background color, border color, and text color based on the selected and visible states. It also adjusts the layout and styling of the input field used for renaming the node.

### CODE REVIEW

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
we have some nodes' text color indistinguishable from background

image

**Comment by SYBIOTE** [comment]
yeah thats intended, for disabled/invisible  nodes, take a look at the jira ticket, i suppose we can bump up the readability a little though

https://tsu.atlassian.net/browse/IR-3216

**Comment by DanielBelmes** [comment]
https://www.figma.com/design/DXwFHGa5g0peqqSjyjpxQz/Studio-Design?node-id=7068-22960&t=HrmEHlB4EQy8V208-4

had paul take a look and he is whipping up a figma test case to ensure we're following the design since a figma wasnt' included. I'm also taking a look but it's nearing the end of my day

**Comment by DanielBelmes** [comment]
#191B1F for background. #42454D for the text color from the designs for disabled. Paul recommended to go into FIgma devmode and just scope out the setup on the right there next to the screen shot to make sure to match.

File: packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx [hunk 1]
```diff
@@ -312,8 +312,14 @@ export const HierarchyTreeNode = (props: HierarchyTreeNodeProps) => {
     <li
       style={props.style}
       className={twMerge(
-        `bg-${props.index % 2 ? 'theme-surfaceInput' : 'zinc-800'}`,
-        selected ? 'border border-gray-100' : 'border-none'
+        `${
```
Comment (HexaField) [inline packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx]: do these values properly update reactively?

File: packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx [hunk 2]
```diff
@@ -312,8 +312,14 @@ export const HierarchyTreeNode = (props: HierarchyTreeNodeProps) => {
...
+          props.index % 2
+            ? `${selected ? 'bg-[#1d1f23]' : 'bg-theme-surfaceInput'} hover:bg-[#1d1f23]`
+            : `${selected ? 'bg-zinc-700' : 'bg-zinc-800'} hover:bg-zinc-700`
+        }`,
+        selected ? 'text-white' : 'text-gray-300',
+        'hover:text-white',
+        `${visible ? '' : 'bg-[#191B1F] text-[#6B7280]'}`
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx]:
```suggestion
        visible && 'bg-[#191B1F] text-[#6B7280]'
```

File: packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx [hunk 3]
```diff
@@ -312,8 +312,14 @@ export const HierarchyTreeNode = (props: HierarchyTreeNodeProps) => {
     <li
       style={props.style}
       className={twMerge(
-        `bg-${props.index % 2 ? 'theme-surfaceInput' : 'zinc-800'}`,
-        selected ? 'border border-gray-100' : 'border-none'
+        `${
+          props.index % 2
+            ? `${selected ? 'bg-[#1d1f23]' : 'bg-theme-surfaceInput'} hover:bg-[#1d1f23]`
+            : `${selected ? 'bg-zinc-700' : 'bg-zinc-800'} hover:bg-zinc-700`
+        }`,
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx]:
```suggestion
          props.index % 2
            ? (selected ? 'bg-[#1d1f23]' : 'bg-theme-surfaceInput'} hover:bg-[#1d1f23]')
            : (selected ? 'bg-zinc-700' : 'bg-zinc-800'} hover:bg-zinc-700'),
```

twMerge works on `any` type. also just the final result needs to be string which in this case already is

**Review by aditya-mitra — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx (hunk 3 above)
Comment (SYBIOTE) [reply packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx]: oh, cool works either way, your suggestion is easier to read, let's go with that

File: packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx (hunk 1 above)
Comment (DanielBelmes) [reply packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx]: It is reactive btw. But the current iteration has some other issues that aren't applying styling and have syntax errors atm.

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
there little visual differences compared to figma
https://www.figma.com/design/DXwFHGa5g0peqqSjyjpxQz/Studio-Design?t=3S55eGfzuPsSseqh-0

**Comment by SYBIOTE** [comment]
implemented changes

**Review by aditya-mitra — APPROVED** [review]

**Pull request merged**